rubix-core/src/main/java/com/qubole/rubix/core/CachedReadRequestChain.java

```diff
@@ -100,6 +103,9 @@ public Integer call()
         log.info(String.format("Read %d bytes from cached file", read));
         fileChannel.close();
         raf.close();
+        if (statistics != null) {
+          statistics.incrementBytesRead(read);
```

Comment:
Is there a way that we can provide our own Statistics implementation? Basically, we would want to differentiate amount of data read locally (i.e. in CachenRRC) and amount of data read from other nodes (i.e. in NonLocalRRC).

Reply:
Yes right now we are only populating bytes read counter for local read. Current implementation of NonLocalRead stats gives you both remote read and non local read combined. But once we have the asynchronous cache warm up feature, it will be easy to differentiate between the data read from the non local cache and data we are downloading in the remote node as part of this request.

Reply:
Do you know how are these statistics shown by Hive jobs? i.e. does it call FileSystem.printStatistics method? Based on that we can think of workarounds

Reply:
FileSystem has the statistics object. This is part of hadoop file system level counter. Hive just gets a report from AM and prints it